### PR TITLE
Soften dependency on public container registries

### LINKED ISSUE #98: Remove dependencies on public container registries
We currently reference a couple images from quay.io and docker.io.  These should be brought into our ACR instance and referenced that way instead.  Update the instructions to include this step and changes.

### PR DESCRIPTION
**Pull request by ckittel**
* Update the GitOps part to use all non mcr-based images from the ACR instance we deploy with the cluster.
* Add specific instructions on how to tighten up the deployment artifacts to support a even more restricted environment.
* Workload is **not** included in this. Once this is validated/approved, then I'll move on to the workload.
* inner-loop shell script wasn't updated (azcli was)
* Couple unrelated formatting/updates as well

Closes #98

### CODE REVIEW

**Comment by ckittel** [comment]
@ferantivero -- I'm not going to merge this just yet, but I'd love your 👀 on this.  I have one more end-to-end test I want to run, but I'll be doing that tomorrow.  But if you have any suggestions on the presentation/etc here, please do start throwing feedback my way as I do the final technical validation.

File: 06-gitops.md
Comment (ferantivero) [inline 06-gitops.md:79]: could we start considering to fork a pre-requisite for all the paths? This way, we could have a first citizen experience on this amazing best practice <3

Ideally in this PR, but I really don't want to block us because of this. A future PR will do as well if that is the team final decision.

Comment (ckittel) [reply 06-gitops.md:79]: Understood.  This non-forked flow still "just works" and in the interest of keeping the barrier to entry low, I think we'll have to schedule time in a future sprint to think about reflowing the document.

**Comment by ckittel** [comment]
Okay, validation complete.  If steps are followed, no more runtime dependencies on public container registries for baseline settings.  Open to additional review feedback before I merge here, so let me know.  Thanks.

File: 06-gitops.md
Comment (neilpeterson) [inline 06-gitops.md:59]: The AKS service itself depends on Microsoft-hosted container registries.

I find myself re-reading this sentence a few times wondering if it's something that I care about, should be concerned about, influences my decisions. 

Does it add value here?

Comment (neilpeterson) [inline 06-gitops.md:59]: Potential suggestion for simplifying this paragraph.

Public container registries are subject to faults such as outages (no SLA) or request throttling. Interruptions like these can be crippling for a system that needs to pull an image _right now_. To minimize the risks of using public registries, consider storing all applicable container images in a registry that you control, such as the SLA-backed Azure Container Registry.

Comment (ckittel) [reply 06-gitops.md]: No, not really.  Was just trying to fend off the "but many core images images still come from public repos, just microsoft-controlled ones" feedback.  But if that happens, that's okay, that's a good conversation to have - and honestly this probably wouldn't have prevented it.

Comment (ckittel) [reply 06-gitops.md]: Updated to a variant of this.  Thanks.  I moved from "consider" to "just do it" type wording in the last sentence.

Comment (ckittel) [reply 06-gitops.md]: Resolving as this paragraph was rewritten and excluded this sentence.

File: 06-gitops.md
Comment (ckittel) [reply 06-gitops.md:79]: (marking resolved -- if you want to reopen, please do so.  Thanks)

File: 10-validation.md
Comment (ferantivero) [inline 10-validation.md:123]: Ideally we could prescribe a step in here or just let it be a note/tip. 

Please let's not be it a blocker, and proceed based on your preference.

Comment (ferantivero) [reply 10-validation.md:123]: Please discard this comment, I'm seeing we created specific steps just below.

**Review by ferantivero — APPROVED** [review]
LGTM 🚢  - again I love this best practice besides the new throttling thing. From security point of view, this could be also beneficial since it is possible scan the images from your registry I think.

File: 05-aks-cluster.md
Comment (hallihan) [inline 05-aks-cluster.md:135]:
```suggestion
:warning: To aid in ease of deployment of this cluster and your experimentation with workloads, Azure Policy is currently configured to allow your cluster to pull images from _public container registries_ such as Docker Hub and Quay. For a production system, you'll want to update the Azure Policy named `pa-allowed-registries-images` in your `cluster-stamp.json` file to only list those container registries that you are willing to take a dependency on and what namespaces those policies apply to. This will protect your cluster from unapproved registries being used, which may prevent issues while trying to pull images from a registry which doesn't provide SLA guarantees for your deployment.
```
Not sure if this suggestion is noticeably clearer.  I'll leave it to your judgement, but the end of the sentence seemed a bit strange.

**Review by hallihan — APPROVED** [review]
Looks good to me, one minor suggestion on wording for the warning sentence in 05-aks-cluster.md

File: 06-gitops.md
Comment (ckittel) [reply 06-gitops.md]: Marking this conversation resolved, however, if you don't like the new version, just let me know and we can tweak it some more.  Thanks!

**Comment by ckittel** [comment]
@ferantivero -- yeah, registry-based scanning is "unlocked" because of this.  Good point!

**Pull request merged**
